### PR TITLE
✅ Add e2e test for `form[custom-validation-reporting="as-you-go"]`

### PR DESCRIPTION
**Pull request by alanorozco**
Adds test case related to #34641, which should fail with a rollforward.

### CODE REVIEW

File: build-system/tasks/e2e/e2e-types.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ const Key = {
   'ArrowLeft': 'ArrowLeft',
   'ArrowRight': 'ArrowRight',
   'ArrowUp': 'ArrowUp',
+  'Backspace': 'Backspace',
```
Comment (alanorozco) [inline build-system/tasks/e2e/e2e-types.js:55]: Necessary to trigger a backspace. I don't like this double mapping :(

File: extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    it('should not send request when submitting default (invalid) input', async () => {
+      const submitButton = await controller.findElement('[type="submit"]');
+      await controller.click(submitButton);
+      await expect(
+        'https://example.com/subscribe?__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
+      ).to.not.have.been.sent;
```
Comment (micajuine-ho) [inline extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js:37]: Curious, this check for a request being sent is natively built into our testing framework?

**Review by micajuine-ho — APPROVED** [review]
Thank you for adding this test Alan!!

**Review by caroqliu — APPROVED** [review]

File: extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js (hunk 2 above)
Comment (alanorozco) [reply extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js:37]: Only on e2e tests.

```
extensions/amp-consent/0.1/test-e2e/test-amp-consent-client-side.js:
  102          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=sufficient'
  103:       ).to.have.been.sent;

extensions/amp-consent/0.1/test-e2e/test-amp-consent-server-side-expire-cache.js:
  47          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=insufficient'
  48:       ).to.have.been.sent;

  92          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=sufficient'
  93:       ).to.have.been.sent;

extensions/amp-consent/0.1/test-e2e/test-amp-consent-server-side.js:
  85          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=insufficient'
  86:       ).to.have.been.sent;

extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js:
  36          'https://example.com/subscribe?__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
  37:       ).to.not.have.been.sent;
  38      });

  48          'https://example.com/subscribe?__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
  49:       ).to.have.been.sent;

extensions/amp-video/0.1/test-e2e/test-amp-video-analytics.js:
  53          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking&id=myVideo'
  54:       ).to.have.been.sent;

  64          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking&id=myVideo2'
  65:       ).to.have.been.sent;

extensions/amp-video/0.1/test-e2e/test-amp-video-flexible-bitrate.js:
  228            'https://amp-dev.cdn.ampproject.org/mbv/s/amp.dev/static/samples/video/tokyo.mp4?amp_video_host_url=https%3A%2F%2Famp.dev%2F&__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
  229:         ).to.have.been.sent;
```

File: build-system/tasks/e2e/e2e-types.js (hunk 1 above)
Comment (estherkim) [reply build-system/tasks/e2e/e2e-types.js:55]: Agreed :(

**Review by estherkim — APPROVED** [review]
e2e changes ✅

**Pull request closed without merging**